
---

### 🚚 **1. Delivery Time (Logistics)** 🚚

You're managing a **logistics company**, and the **average delivery time** for packages is **3 days** with a **standard deviation** of **0.5 days**. You want to know the **probability** that a package will be delivered in **less than 2.5 days**. 📦⏳


---


In [ ]:
#### Python Code:
mean = 3             # Average delivery time
std_dev = 0.5        # Standard deviation OF TIME
x = 2.5              # Time to check probability for

# less than 2.5
# this function always calculate probability on left side so 
# if it was more than 2.5 then we need to give x=-2.5
from scipy import stats
probability = stats.norm.cdf(x, mean, std_dev)# population parameters

print(f"Probability of delivery in less than 2.5 days: {probability:.4f}")

Probability of delivery in less than 2.5 days: 0.1587



#### **Interpretation:**

This shows you the **15 % (0.1587) of chance** that a package will arrive **in less than 2.5 days**. 📦🎉





Now, you asked about **`ppf`**:  
> 🔹 **`ppf`** is the **Percent Point Function** (also called the *inverse* of the CDF).  
> 🔹 It tells you: for a given probability, what is the corresponding delivery time?

**Example:**  
Suppose you want to know:  
_"How fast must a delivery happen to be faster than 10% of all deliveries?"_

You would use `ppf(0.1, mean, std_dev)`.




✅ This finds the delivery time where **only 10%** of deliveries are **faster** than that time.



In [4]:

# Find the delivery time that corresponds to the 10th percentile
percentile = 0.15
delivery_time = stats.norm.ppf(percentile, mean, std_dev)

print(f"Delivery time at the 10th percentile: {delivery_time:.2f} days")


Delivery time at the 10th percentile: 2.48 days
